apps/namadillo/src/App/Transactions/TransactionCard.tsx

```diff
@@ -86,25 +86,25 @@ const getTransactionInfo = (
 
   const parsed = typeof tx.data === "string" ? JSON.parse(tx.data) : tx.data;
   const sections: RawDataSection[] = Array.isArray(parsed) ? parsed : [parsed];
+  const target = sections.find((s) => s.targets?.length);
+  const source = sections.find((s) => s.sources?.length);
 
-  let receiver: string | undefined;
   let amount: BigNumber | undefined;
+  let receiver: string | undefined;
 
-  // Find first section with targets or sources that has amount
-  const targetSection = sections.find((sec) => sec.targets?.[0]?.amount);
-  const sourceSection = sections.find((sec) => sec.sources?.[0]?.amount);
-
-  if (targetSection?.targets?.[0]) {
-    amount = new BigNumber(targetSection.targets[0].amount);
-    receiver = targetSection.targets[0].owner;
+  if (target?.targets) {
+    const mainTarget = target.targets.reduce((max, cur) =>
+      new BigNumber(cur.amount).isGreaterThan(max.amount) ? cur : max
+    );
+    amount = new BigNumber(mainTarget.amount);
+    receiver = mainTarget.owner;
   }
-
-  if (!amount && sourceSection?.sources?.[0]) {
-    amount = new BigNumber(sourceSection.sources[0].amount);
+  // fall back to sources only when we had no targets
+  if (!amount && source?.sources?.[0]) {
+    amount = new BigNumber(source.sources[0].amount);
   }
 
-  // Find sender from any section with sources
-  const sender = sections.find((sec) => sec.sources?.[0]?.owner)?.sources?.[0]
+  const sender = sections.find((s) => s.sources?.[0]?.owner)?.sources?.[0]
```

Comment:
shouldn't we move the `?.sources?.[0]` inside the sender? could the sender be still undefined?
```
  const sender = sections.find((s) => s.sources?.[0]?.owner?.sources?.[0])
```

I didn't test on real world, only thinking about the pure js logic

Reply:
There's always a sender. 

So if I did: 

```  const sender = sections.find((s) => s.sources?.[0]?.owner?.sources?.[0])
```

I'd still have to write sender?.source?.[0]?.owner to extract the value, so that's what I'm doing above but shorthand. Make sense?

Reply:
I just mean, if we add the `?.` on `sources?.[0]`, we are expecting that, in a border case, we could have `sources === undefined`

so, if this is the case, should we consider another `s.sources?.[0]?.owner` to be searched?

I do understand that we do not expect this on happy paths, but if we are 100% sure about all cases, we could simply remove all `?`, no? As we don't, we could be aligned on code on the same idea